### DISCUSSION TITLE
Some fields (like 'merged_at', 'base', 'head') are absent in the search results for pull requests

### DISCUSSION THREAD

**Discussion by denis-bigbrassband-com** [community/community#24491] [Apps, API and Webhooks]
<p>Hi everyone,</p>
<p>We are using ‘/search/issues’ API v3 call to get a list of recently changed pull requests:<br>
<code>https://api.github.com/search/issues?q=repo:&lt;user&gt;/&lt;repo&gt;+is:pr+updated:%3E2020-07-11T23:57:00Z&amp;page=1&amp;sort=updated&amp;order=asc</code></p>
<p>But unlike the ‘/repos///pulls’ API v3 calls the result of the ‘search’ call does not contain some fields, for example, “merged_at”, “head”, “base”.</p>
<p>With the ‘search’ call we tried to reduce the traffic to the Github server (we use it instead of getting all available requests with ‘/repos///pulls’). But it seems that we are unable to get all the required information about pull requests with the ‘search’ call.<br>
Are there any options to get these fields in the ‘search’ calls? Or are there any other solutions to get the recently changed pull requests?</p>
<p>Thank you.</p>

**Comment by francisfuzz** [comment, marked as answer]
<p><a class="mention" href="/denis-bigbrassband-com">@denis-bigbrassband-com</a>: The GraphQL API also exposes a <a href="https://docs.github.com/en/graphql/reference/queries#searchresultitemconnection"><code>search</code></a> field, where <code>ISSUE</code> is one of its supported <a href="https://docs.github.com/en/graphql/reference/enums#searchtype">SearchType</a>s.</p>
<p>Here’s an example query that searches for all pull requests in the <code>github/linguist</code> repository updated after a certain time (you can modify this example to your liking):</p>
<pre><code class="lang-auto">query {
  search(type: ISSUE, first: 10, query: "repo:github/linguist is:pr updated:&gt;2020-07-11T23:57:00Z") {
    edges {
      node {
        ... on PullRequest {
          author {
            login
          }
          number
          mergedAt
          baseRefName
          headRefName
        }
      }
    }
  }
}
</code></pre>
<details>
<summary>Result Set</summary>
<pre><code class="lang-json">{
  "data": {
    "search": {
      "edges": [
        {
          "node": {
            "author": {
              "login": "mortend"
            },
            "number": 4938,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "uno-color"
          }
        },
        {
          "node": {
            "author": {
              "login": "PieterO"
            },
            "number": 4937,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "add-velocity-template-language"
          }
        },
        {
          "node": {
            "author": {
              "login": "Alhadis"
            },
            "number": 4936,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "generic-extensions"
          }
        },
        {
          "node": {
            "author": {
              "login": "jdbender66"
            },
            "number": 4935,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "patch-1"
          }
        },
        {
          "node": {
            "author": {
              "login": "sheerun"
            },
            "number": 4934,
            "mergedAt": "2020-07-29T08:29:42Z",
            "baseRefName": "master",
            "headRefName": "bundler"
          }
        },
        {
          "node": {
            "author": {
              "login": "sheerun"
            },
            "number": 4933,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "carp"
          }
        },
        {
          "node": {
            "author": {
              "login": "sheerun"
            },
            "number": 4932,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "asl"
          }
        },
        {
          "node": {
            "author": {
              "login": "Alhadis"
            },
            "number": 4931,
            "mergedAt": "2020-07-28T09:26:30Z",
            "baseRefName": "master",
            "headRefName": "misc-fixes"
          }
        },
        {
          "node": {
            "author": {
              "login": "brnrdog"
            },
            "number": 4929,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "reason-bs-8.1"
          }
        },
        {
          "node": {
            "author": {
              "login": "sambacha"
            },
            "number": 4928,
            "mergedAt": null,
            "baseRefName": "master",
            "headRefName": "add-edi-dev0"
          }
        }
      ]
    }
  }
}
</code></pre>
</details>
<p>With GraphQL, you can specify the <a href="https://docs.github.com/en/graphql/guides/migrating-from-rest-to-graphql#example-getting-the-data-you-need-and-nothing-more">data you need and nothing more</a>: this query only targets five fields per resulting pull request. This kind of configurability is what sets it apart from the REST API’s preconfigured responses.</p>
<p>Does this help with what you’re looking to accomplish?</p>

**Comment by denis-bigbrassband-com** [comment]
<p>Hi <a class="mention" href="/francisfuzz">@francisfuzz</a>,</p>
<p>Thank you for the recommendations.<br>
We currently don’t use GraphQL in our project. It seems that it is time to start to do this. Could you recommend any Java library to work with GraphQL API?</p>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/denis-bigbrassband-com">@denis-bigbrassband-com</a> - Thanks for writing back! I don’t have any official recommendations for a specific Java library to work with our GraphQL API––I suggest opening a new topic in our <a href="https://github.community/c/programming-help-and-discussion/39">Programming Help and Discussion category</a> and raising that question there.</p>
<p>If it helps as an additional resource, I highly encourage you to check out this <a href="https://developer.github.com/v3/libraries/">GitHub Developer page showcasing Octokit (our official libraries for interfacing with the GitHub API) as well as other third-party libraries</a>.</p>
<p>Does that help?</p>

**Comment by denis-bigbrassband-com** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a><br>
Thank you. I experimenting with this a bit, and it seems this gives all the required information for us.<br>
Marked as resolved. Thank you again.</p>
